### ISSUE TITLE
Week 6 ready for review

### ISSUE THREAD

**Issue by dwellons** [dwellons/DonationLog#5]
Hello @pawaitemadisoncollege, week 6 is ready for you to review!

http://donationlog.us-east-2.elasticbeanstalk.com/

What were your key learning points/takeaways?
That the documentation that they provide seems overwhelming at first, but it was actually pretty helpful. Following it can help you.

What challenged you, what problems did you solve and what resources did you use to solve them?

Just initially setting up the environment/application. I got an error "the instance profile "aws-elasticbeanstalk-ec2-role associated with the environment does not exist. I figured it had something to do with the setup so I tried to delete the environment and start over, but it took me a minute to find that you have to remove the application and it will remove the environment as a result. I found in the setup where you choose key pairs and an instance profile to use with ec2, so I went through the instructions to set both of those up and then tried again. The application started but my page wasn't loading.

I had an error on the page relating the taglib URL in the start of my index file. That started a whole different journey, which led me to find that my maven and JSTL dependencies were not setup correctly. I had two war files under my target directory being created, one named DonationLog and the other user display exercise. All of my work going into the user display exercise one, and I did not upload that one to AWS. 

I cleaned that up and got rid of the extra old files. I was still getting the error for the jstl link at the start of my index page. in IntelliJ the link was red and had an error, and when I clicked the message, it brought up something about dependencies on the side. I looked through my xml file and there didn’t seem to be any for jstl tags, so I searched for jstl dependencies and found that I didn’t have any jstl files in my webapp/lib directory at all. I downloaded those from oracle and followed the instructions to set it up and searched for a dependency for jstl to add to the xml file. 

After all of that it was displaying. I moved on to the database and once I found where amazon had moved things to on their website it wasn't bad. I have been using phpMyAdmin for database management, that wasn't going to work for this, so I installed MySQL workbench and got that connected to the database and imported everything. I replaced the URL and username/password in my database.properties file but not the hibernate config file and got an error on the webpage relating to hibernate and I remembered I had to change it there too. I changed it there and my database is connected, and it is displaying.

the links from my sidebar do not work anymore on aws so I’m going to have to look at that.

Resources Used: 
JSTL

https://www.tutorialspoint.com/jsp/jsp_standard_tag_library.htm

Maven JSTL Dependency

https://mvnrepository.com/artifact/javax.servlet/jstl/1.2

<img width="1782" alt="FirstAWSDeployment" src="https://github.com/dwellons/DonationLog/assets/143457620/c454f3e6-824d-49ae-979a-69f00531aff6">

**Comment by pawaitemadisoncollege** [comment]
Hey @dwellons! I realize I'm reviewing this out of order, but I was pretty pumped that you'd finished week 6 already  so I just had to take a look at this! Sounds like you got in some good debugging/troubleshooting practice and definitely won the game there! 

Regarding the links - hopefully checking out the url in the browser when you click on a broken link will help you figure out what might be missing (or extra!) in the link.  I've seen that kind of behavior if my links weren't "relative" enough if that makes any sense.

Keep up the great momentum! Looks like you may have hit checkpoint 2 with this work, which isn't due until next week.

**Comment by dwellons** [comment]
@pawaitemadisoncollege I have been hoping to get a little ahead so that is good to hear! I will be out of town Monday and Tuesday next week, I will have my computer but not sure if I will have time for school

**Comment by dwellons** [comment]
@pawaitemadisoncollege after the troubleshooting yesterday here is my new AWS link!
http://donationlog-env.eba-2mr56wsp.us-east-2.elasticbeanstalk.com/

**Comment by pawaitemadisoncollege** [comment]
Awesome - hope this deployment solves that cost issue you observed.   

- [ ] Be sure to add that link to the list of individual projects in the student repo: https://github.com/mad-ent-java-s24/student/blob/main/IndividualProjects.md so the link is easy to find.

- [ ] I noticed the issue with the local links versus aws links when I was testing things out. It looks like the local links include the application context root (war file name) while this is not needed on aws. I wonder if using more relative links throughout the application would solve this (I did not poke around at your code at all, that's just the first thing that occurred to me as I was looking at it.
